saleor/product/utils/variants_picker.py

```diff
@@ -37,29 +49,179 @@ def _attributes_to_map(instance: ProductVariant) -> Dict[str, str]:
     return attribute_map
 
 
+@dataclass
+class VariantValues:
+    pk: str
+    name: str
+    slug: str
+
+
+@dataclass
+class VariantAttributes:
+    pk: str
+    name: str
+    slug: str
+    values: List[VariantValues]
+
+    def as_dict(self):
+        return {
+            "pk": self.pk,
+            "name": self.name,
+            "slug": self.slug,
+            "values": [asdict(v) for v in self.values],
+        }
+
+
+# TODO: probably we need price dataclass anyway - investigate later
+@dataclass
+class MoneyAsDict:
+    currency: str
+    gross: Decimal
+    grossLocalized: Decimal
+    net: Decimal
+    netLocalized: Decimal
+
+
+@dataclass
+class MoneyRangeAsDict:
+    minPrice: Optional[MoneyAsDict]
+    maxPrice: Optional[MoneyAsDict]
+
+    def as_dict(self) -> dict:
+        return {
+            "minPrice": asdict(self.minPrice) if self.minPrice else None,
+            "maxPrice": asdict(self.maxPrice) if self.maxPrice else None,
+        }
+
+
+@dataclass
+class VariantData:
+    id: str
+    availability: bool
+    price: Optional[MoneyAsDict]
+    priceUndiscounted: Optional[MoneyAsDict]
+    attributes: AttributesMapType
+    priceLocalCurrency: Optional[MoneyAsDict]
+    schemaData: dict  # TODO: Can it be changed to schema?
+
+    def as_dict(self):
+        price = asdict(self.price) if self.price else None
+        priceLocalCurrency = (
+            asdict(self.priceLocalCurrency) if self.priceLocalCurrency else None
+        )
+        priceUndiscounted = (
+            asdict(self.priceUndiscounted) if self.priceUndiscounted else None
+        )
+        return {
+            "id": self.id,
+            "availability": self.availability,
+            "price": price,
+            "priceUndiscounted": priceUndiscounted,
+            "attributes": self.attributes,
+            "priceLocalCurrency": priceLocalCurrency,
+            "schemaData": self.schemaData,
+        }
```

Comment:
As above, I think it's not necessary and could be handled be default values in fields definitions.